packages/itwinui-react/src/core/Badge/Badge.tsx

```diff
@@ -69,15 +48,34 @@ export const Badge = (props: BadgeProps) => {
 
   useTheme();
 
-  const _style = backgroundColor
-    ? {
-        '--iui-badge-background-color': getBadgeColorValue(backgroundColor),
-        ...style,
-      }
-    : { ...style };
+  // choosing 'primary' status should result in data-iui-status equaling 'informational'
+  const reducedBackgroundColor =
+    backgroundColor === 'primary' ? 'informational' : backgroundColor;
+
+  const statuses = ['informational', 'positive', 'negative', 'warning'];
+
+  const isStatus =
+    reducedBackgroundColor && statuses.includes(reducedBackgroundColor);
+
+  const _style =
+    reducedBackgroundColor && !isStatus
+      ? {
+          '--iui-badge-background-color': isSoftBackground(
+            reducedBackgroundColor,
+          )
+            ? SoftBackgrounds[reducedBackgroundColor]
+            : reducedBackgroundColor,
```

Comment:
i dont think it is necessary to pass status colors as css variables. they should be automatically set when using the data attribute

so most of this logic can be simplified and moved into two simple helper functions that you can call like this:

```
'--iui-badge-background-color': getBadgeColorValue(backgroundColor)
```
```
data-iui-status={getStatusValue(backgroundColor)}
```

Reply:
This has changed. I have it so that the background color is set in style regardless of whether `backgroundColor` is a status or not.

Reply:
Of course! Thanks for the explanation. This has been changed

as for `reducedBackgroundColor`, `'primary'` is not an option for `data-iui-status`, so the change from `'primary'` to `'informational'` is still necessary